chain/src/actors/syncer/actor.rs

```diff
@@ -866,19 +828,11 @@ impl<B: Blob, R: Rng + Spawner + Metrics + Clock + GClock + Storage<B>, I: Index
                                     // Persist the block
                                     debug!(?digest, height = block.height, "received block");
                                     let _ = response.send(true);
-                                    blocks
-                                        .put(block.height, digest, value)
+                                    self.blocks
+                                        .put(block.height, digest, block)
                                         .await
                                         .expect("Failed to insert finalized block");
 
-                                    // Notify waiters
-                                    if let Some(waiters) = waiters.remove(&digest) {
```

Comment:
We no longer notify waiters (managed by `broadcast::buffered`) if we get a useful request out-of-band. We should never be making requests for things we'd be waiting on broadcast buffered for, so this is ok.